webrob/pages/admin/db.py

```diff
@@ -145,3 +154,50 @@ def db_remove_route(table):
     if data is not None and 'id' in data and db_find(cls, data['id']):
         db_remove(cls, data['id'])
     return jsonify(result=None)
+
+
+@app.route('/db/page/get_neem_hub_settings')
+@admin_required
+def render_neem_hub_settings_page_get():
+    # PasswordForm used for validating given password field
+    form = PasswordForm()
+    neemHubSettings = get_settings()
+    if neemHubSettings is not None:
+        return render_template('admin/neem_hub_settings.html', form=form, neemHubSettings = get_settings())
+    else:
+        return render_template('admin/neem_hub_settings.html', form=form, neemHubSettings=NEEMHubSettings())
+
+@app.route('/db/page/post_neem_hub_settings', methods=['POST'])
+@admin_required
+def render_neem_hub_settings_post():
+    app.logger.info('render neem hub settings post method.... ')
+    req = request.form
+    if req is not None:
+        neemHubSettings = get_settings()
+        if neemHubSettings is None:
+            neemHubSettings = NEEMHubSettings()
+        neemHubSettings.MONGO_HOST = req.get("MONGO_HOST")
+        neemHubSettings.MONGO_PORT = req.get("MONGO_PORT")
+        neemHubSettings.MONGO_USER = req.get("MONGO_USER")
+        neemHubSettings.MONGO_DB = req.get("MONGO_DB")
+        neemHubSettings.MONGO_PASS = req.get("MONGO_PASS")
+
+        db.session.add(neemHubSettings)
+        db.session.commit()
+        app.logger.info('Configuration has been stored!')
+        mongoDBMetaCollection = getMongoDBMetaCollection(neemHubSettings)
+
+        if mongoDBMetaCollection is None:
+            app.logger.error('------------ mongoDb connection can not be created ------------')
+            flash('Failure connecting with mongodb with given credentials, please check inputs!', "warning")
+            return redirect(url_for('render_neem_hub_settings_page_get'))
+        else:
+            # if connection is secured then update neem_ids from mongodb meta collection so that neem discovery page has latest updates
+            neem_manager.__set_neem_ids__()
```

Comment:
why starts this method with two underscore?

Reply:
no apparent reason. the previous method __get_neem_ids__ was renamed as __set_neem_ids__
I have removed the '__' from method name.